src/driver/tests/experiment_driver_test.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for experiment_driver.py."""
+
+from absl.testing import absltest
+import numpy as np
+import pandas as pd
+from tempfile import TemporaryDirectory
+from unittest.mock import patch
+
+from wfa_planning_evaluation_framework.data_generators.data_design import DataDesign
+from wfa_planning_evaluation_framework.data_generators.synthetic_data_design_generator import (
+    SyntheticDataDesignGenerator,
+)
+from wfa_planning_evaluation_framework.data_generators import simple_data_design_example
+from wfa_planning_evaluation_framework.driver.experiment_driver import ExperimentDriver
+from wfa_planning_evaluation_framework.driver.experimental_trial import (
+    ExperimentalTrial,
+)
+from wfa_planning_evaluation_framework.driver import sample_experimental_design
+
+
+class FakeExperimentalTrial(ExperimentalTrial):
+    def evaluate(self, rng: np.random.Generator):
+        return pd.DataFrame(
+            {
+                "data_set_name": [self._data_set_name],
+                "trial_descriptor": [str(self._trial_descriptor)],
+            }
+        )
+
+
+class ExperimentDriverTest(absltest.TestCase):
+    def test_sample_experimental_design(self):
+        sample_design = sample_experimental_design.generate_experimental_design_config(
+            np.random.default_rng(seed=1)
+        )
+        self.assertLen(list(sample_design), 100)
+
+    @patch(
+        "wfa_planning_evaluation_framework.driver.experiment.ExperimentalTrial",
+        new=FakeExperimentalTrial,
+    )
+    def test_experiment_driver_with_sample_experimental_design(self):
+        with TemporaryDirectory() as d:
+            data_design_dir = d + "/data"
+            output_file = d + "/output"
+            intermediate_dir = d + "/intermediates"
+            data_design_generator = SyntheticDataDesignGenerator(
+                data_design_dir, 1, simple_data_design_example.__file__, False
+            )
+            data_design_generator()
```

Comment:
Got error message in this line as shown below:

Running tests under Python 3.8.10: /Users/riemanli/.virtualenvs/pef_env/bin/python3
[ RUN      ] ExperimentDriverTest.test_experiment_driver_with_sample_experimental_design
[  FAILED  ] ExperimentDriverTest.test_experiment_driver_with_sample_experimental_design
[ RUN      ] ExperimentDriverTest.test_sample_experimental_design
[       OK ] ExperimentDriverTest.test_sample_experimental_design
======================================================================
ERROR: test_experiment_driver_with_sample_experimental_design (__main__.ExperimentDriverTest)
ExperimentDriverTest.test_experiment_driver_with_sample_experimental_design
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/mock.py", line 1325, in patched
    return func(*newargs, **newkeywargs)
  File "experiment_driver_test.py", line 63, in test_experiment_driver_with_sample_experimental_design
    data_design_generator()
  File "/Users/riemanli/.virtualenvs/pef_env/lib/python3.8/site-packages/wfa_planning_evaluation_framework-0.1-py3.8.egg/wfa_planning_evaluation_framework/data_generators/synthetic_data_design_generator.py", line 105, in __call__
    for data_set_parameters in self._fetch_data_design_config():
  File "/Users/riemanli/.virtualenvs/pef_env/lib/python3.8/site-packages/wfa_planning_evaluation_framework-0.1-py3.8.egg/wfa_planning_evaluation_framework/data_generators/synthetic_data_design_generator.py", line 115, in _fetch_data_design_config
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 844, in exec_module
  File "<frozen importlib._bootstrap_external>", line 980, in get_code
  File "<frozen importlib._bootstrap_external>", line 1037, in get_data
NotADirectoryError: [Errno 20] Not a directory: '/Users/riemanli/.virtualenvs/pef_env/lib/python3.8/site-packages/wfa_planning_evaluation_framework-0.1-py3.8.egg/wfa_planning_evaluation_framework/data_generators/simple_data_design_example.py'

----------------------------------------------------------------------
Ran 2 tests in 0.084s

FAILED (errors=1)

Any idea?

Reply:
Resolved the issue by following steps:
1. Add PYTHONPATH to _.bash_profile_ or _.zshrc_ file as following:

```
PYTHONPATH=$HOME/dir_which_contains_planning_evaluation_framework_repo:.
export PYTHONPATH
```

2. Create a Symlink named “wfa_planning_evaluation_framework” with command:
	
```
ln -s prefix_path_to/planning-evaluation-framework/src/ dir_which_contains_planning_evaluation_framework_repo/wfa_planning_evaluation_framework
```